coding-the-legislation/40_legislation_evolutions.md

```diff
@@ -51,76 +53,111 @@ we get the output:
     flat_tax_on_salary:
         2016-01: 500
         2015-12: 400
-        ...
         2015-01: 400
         2014-12: 440
-        ...
         2014-01: 440
 ```
 
 [Read more about how to code parameters](./legislation_parameters.md#parameters-and-time).
 
-## Variable defined only for a specific time intervale
+## Formula evolving over time
+
+Some fiscal or benefit mechanism significantly evolve over time, with bigger changes than a simple parameter adjustement.
 
-As the legislation evolves, some fiscal or benefit mechanism appear and disapear. When creating a variable, you can specify the attribute `start_date` and `stop_date` to defines in which time intervale this variable make sense.
+For instance, let's assume that from the 1st of Jan. 2017, our previous `flat_tax_on_salary` will not be applied on the first `1000` earned by the person.
 
-When called outside of its definition intervale, a variable will **not** execute its formula and instead **return its default value**.
+We implement this rule by adding a new formula to our variable, and _dating_ it:
 
-For instance:
 ```py
 class flat_tax_on_salary(Variable):
     column = FloatCol
     entity = Person
     label = u"Individualized and monthly paid tax on salaries"
-    start_date = date(2014, 01, 01)
     definition_period = MONTH
 
-    def function(person, period, legislation):
-        ...
+    def formula_2017(self, simulation, period):
```

Comment:
Sure, but that would also be heavier, as we would have two big python hunks. The original formula is already at the top of the page.